pkg/test/e2e/nodeport-proxy/services.go

```diff
@@ -105,30 +113,106 @@ func (n *ServiceJig) newNamespaceTemplate() *corev1.Namespace {
 	return ns
 }
 
-func (n *ServiceJig) newServiceTemplate(name, namespace string,
+func (n *ServiceJig) newServiceTemplateNodePort(labels map[string]string) corev1.ServiceSpec {
+	ports := corev1.ServiceSpec{
+		Selector: labels,
+		Type:     corev1.ServiceTypeNodePort,
+		Ports: []corev1.ServicePort{
+			{
+				Protocol:   corev1.ProtocolTCP,
+				Port:       80,
+				TargetPort: intstr.FromInt(8080),
+			},
+		},
+	}
+	return ports
+}
+
+func (n *ServiceJig) newServiceTemplateHttps(labels map[string]string) corev1.ServiceSpec {
+	podSpec := corev1.ServiceSpec{
+		Selector: labels,
+		Type:     corev1.ServiceTypeClusterIP,
+		Ports: []corev1.ServicePort{
+			{
+				Protocol:   corev1.ProtocolTCP,
+				Port:       8443,
+				TargetPort: intstr.FromInt(8443),
+			},
+		},
+	}
+	return podSpec
+}
+
+func (n *ServiceJig) newServiceTemplateCommon(name, namespace string,
 	annotations, labels map[string]string) *corev1.Service {
+
+	var podSpec corev1.ServiceSpec
+	if n.Https == true {
+		podSpec = n.newServiceTemplateHttps(labels)
+	} else {
+		podSpec = n.newServiceTemplateNodePort(labels)
+	}
+
 	svc := &corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        name,
 			Namespace:   namespace,
 			Annotations: annotations,
 			Labels:      labels,
 		},
-		Spec: corev1.ServiceSpec{
-			Selector: labels,
-			Ports: []corev1.ServicePort{
-				{
-					Protocol:   corev1.ProtocolTCP,
-					Port:       80,
-					TargetPort: intstr.FromInt(8080),
+		Spec: podSpec,
+	}
+	return svc
+}
+
+func (n *ServiceJig) newRCTemplateNodePort() []corev1.Container {
+	containers := []corev1.Container{
+		{
+			Name:  "agnhost",
+			Image: AgnhostImage,
+			Args:  []string{"netexec", "--http-port=8080"},
+			ReadinessProbe: &corev1.Probe{
+				PeriodSeconds: 3,
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Port: intstr.FromInt(8080),
+						Path: "/",
+					},
 				},
 			},
 		},
 	}
-	return svc
+	return containers
+}
+
+func (n *ServiceJig) newRCTemplateHttps() []corev1.Container {
+	containers := []corev1.Container{
+		{
+			Name:  "agnhost",
+			Image: AgnhostImage,
+			Args:  []string{"netexec", "--http-port=8443", "--tls-cert-file=/localhost.crt", "--tls-private-key-file=/localhost.key"},
+			ReadinessProbe: &corev1.Probe{
+				PeriodSeconds: 3,
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Port: intstr.FromInt(8443),
```

Comment:
For HTTPS you need to specify the scheme: https://github.com/kubernetes/kubernetes/issues/18226#issuecomment-284821545